### ISSUE TITLE
Non-author contributor roles

### ISSUE THREAD

**Issue by JGilbert-eLife** [substance/texture#1263]
# Description

Designates the kind of non-author contributor e.g. reviewer, reviewing editor, senior editor, translator etc.

# User stories

### Author
- As an author, I want to be able to see the role assigned to each non-author contributor so that I can check that this information is correct.

### Production staff
- As production staff, I want to be able to edit the role for a non-author contributor so that I can correct any errors.
- As production staff, I want to be able to assign a role to a non-author contributor from a preset list of options so that I can include this information easily.
- As production staff, I want to be able to add a custom role to a non-author contributor so that I can assign a piece of non-standard information that is not available as a preset option.

# But what if . . . ?

# Consideration

- eLife only allows the following values for `<role>`:
  - Reviewer
  - Senior Editor
  - Reviewing Editor
- Érudit allows e.g. Translator.
- The same person can be both senior and reviewing editor, and at the moment eLife would list that person twice. However, we would prefer simply to add multiple roles per person.
- In eLife articles, there should only be one Reviewing Editor and one Senior Editor per article; however, not every article will have editors listed (see #47).

# XML requirements

Roles are identified using the `role` element.
```
<contrib-group content-type="peer-review">
      <contrib>
            <name>
                  <surname>...</surname>
                  <given-names>...</given-names>
             </name>
            <role>Reviewing Editor</role>
            <xref ref-type="aff" rid="aff5">5</xref>
      </contrib>
      <contrib>
            <name>
                  <surname>...</surname>
                  <given-names>....</given-names>
            </name>
            <role>Senior Editor</role>
            <xref ref-type="aff" rid="aff5">5</xref>
      </contrib>
      <contrib>
            <name>
                  <surname>...</surname>
                  <given-names>....</given-names>
            </name>
            <role>Reviewer</role>
            <xref ref-type="aff" rid="aff6">6</xref>
      </contrib>
      <aff id="aff5"><institution>...</institution><country>...</country></aff>
      <aff id="aff6"><institution>...</institution><country>...</country></aff>
</contrib-group>
<contrib-group content-type="translators">
 	<contrib>
                    <name>
                        <surname>...</surname>
                        <given-names>....</given-names>
                    </name>
                    <role>French Translator</role>
                    <xref ref-type="aff" rid="aff7">7</xref>
 	</contrib>
 	<aff id="aff7">
 	 	<institution>...</institution>
 	 	<country>...</country>
         </aff>
</contrib-group>
```

A contributor may have more than one `role`:
```
<contrib-group content-type="peer-review">
      <contrib>
            <name>
                  <surname>Bright</surname>
                  <given-names>John</given-names>
            </name>
            <role>Reviewing Editor</role>
            <role>Senior Editor</role>
            <xref ref-type="aff" rid="aff5">5</xref>
      </contrib>
      ...
</contrib-group>
```

### Custom Vocabulary
Texture needs to allow for customisable `role` values. There should be a certain set of 'allowed' role values, which a publisher can add to or remove from in their configuration. This will allow for various customisable roles to be accounted for without the need for development work.

### Custom Mandatory configuration
Whether roles are mandatory (or not) should also be customisable in Texture. For example, eLife specify that every peer-review contributor requires a role. However, Érudit may not require a role to be mandated for translators.

# Mock ups

# Proposal

**Comment by michael** [comment]
Thanks for the requirements on this. Very clear and make sense!

Just a few questions:

- is the contrib group type 'peer-review' generic enough?
- will we have a fixed number of contrib group types (authors, peer-review, translators) or do we need to have that extensible? 
- Alternatively: could we just have two groups (authors, and 'other')?
- Custom Vocabulary: It would be good to have the same vocabulary for all roles, otherwise we would need to distinguish between AuthorRoles and NonAuthorRoles is that ok? Are we going to consider the CRedIT taxonomy in the MVP or is that for later?

**Comment by fabiobatalha** [comment]
I would say those 3 groups are enough, and maybe change the name of **peer-review** to **editorial**.

The CRedIT taxonomy doesn't address the translators, I would say it is ok to include this taxonomy but it is still important to allow to include custom values in the element ``role``.

**Comment by Melissa37** [comment]
>is the contrib group type 'peer-review' generic enough?
will we have a fixed number of contrib group types (authors, peer-review, translators) or do we need to have that extensible?

>I would say those 3 groups are enough, and maybe change the name of peer-review to editorial.

I did see/hear one option at one point of having a non-author group. Perhaps that could be used? "non-author" contributor group type, so you'd have `author` and `non-author`, and then the role is used as a display term for the custom vocabulary

>Custom Vocabulary: It would be good to have the same vocabulary for all roles, otherwise we would need to distinguish between AuthorRoles and NonAuthorRoles is that ok? 

Would my suggestions above work to that effect?

>Are we going to consider the CRedIT taxonomy in the MVP or is that for later?

That is a requirement for MVP, we're just waiting on some more input from the community. There will be a separate ticket for this, but for reference here, this is the required tagging:

```
<role vocab="credit" vocab-identifier="http://dictionary.casrai.org/Contributor_Roles" vocab-term="Investigation" vocab-term-identifier="Investigation">Data Collection</role>
```
This issue for eLife is that we use contributions that are not part of the Credit taxonomy, but would like to have a way in the attributes to distinguish those from the roles we're describing in this ticket. I hope that makes sense? If we don't hear from JATS4R we'll have to do something random ourselves - but I'd rather it was community-led.

**Comment by pigeonm** [comment]
At Erudit we only use one type of non author contributor: the translator (but our schema support all of the types listed in the [MARC21 list for relators](http://www.loc.gov/marc/relators/relacode.html))
